### DISCUSSION TITLE
Why different versions of rdkit give different results for MolFromSmiles.

### DISCUSSION THREAD

**Discussion by yangzhao1230** [rdkit/rdkit#6350] [Q&A]
I have the following code:
'''
import rdkit
from rdkit import Chem

print(rdkit.__version__)

smiles = '[NH4+].[NH4+].[Cl-][Pt++]([Cl-])([Cl-])[Cl-]'
mol = Chem.MolFromSmiles(smiles)

'''
I test it with two different versions of rdkit and get different results.
1：2020.09.1
<rdkit.Chem.rdchem.Mol object at 0x148fc81f8d50>

2：2022.09.1
[18:19:20] Explicit valence for atom # 2 Cl, 2, is greater than permitted
None

**Comment by AndreyMozzhegorov** [comment]
This smiles seems broken for me
Also, it's a little bit stale but I tested it in few versions so you don't have to.
Exact code:
```python
import rdkit
from rdkit import Chem
print(rdkit.__version__)

smiles = '[NH4+].[NH4+].[Cl-]Pt++([Cl-])[Cl-]'
mol = Chem.MolFromSmiles(smiles)
print(mol)
```

In 2020.09.1 this doesn't create mol object and any error messages:
```
2020.09.1
None
```

In 2022.09.5 and newer some error logs appear
```
2022.09.5
None
[08:42:18] SMILES Parse Error: syntax error while parsing: [NH4+].[NH4+].[Cl-]Pt++([Cl-])[Cl-]
[08:42:18] SMILES Parse Error: Failed parsing SMILES '[NH4+].[NH4+].[Cl-]Pt++([Cl-])[Cl-]' for input: '[NH4+].[NH4+].[Cl-]Pt++([Cl-])[Cl-]'
```
